packages/core/src/Entity.ts

```diff
@@ -193,11 +217,17 @@ export class Entity extends EngineObject {
    * Create a entity.
    * @param engine - The engine the entity belongs to
    */
-  constructor(engine: Engine, name?: string) {
+  constructor(engine: Engine, name?: string, ...components: ComponentConstructor[]) {
     super(engine);
     this.name = name;
-    this.transform = this.addComponent(Transform);
-    this._inverseWorldMatFlag = this.transform.registerWorldChangeFlag();
+    for (let i = 0, n = components.length; i < n; i++) {
+      const type = components[i];
+      if (!(type.prototype instanceof Transform) || !this.transform) {
+        this.addComponent(type);
+      }
+    }
```

Comment:
```typescript
for (let i = 0, n = components.length; i < n; i++) {
  const type = components[i];
  if (type.prototype instanceof Transform) {
    this._transform || this.addComponent(type);
  } else {
    this.addComponent(type);
  }
}
this._transform || this.addComponent(Transform);
```

This way, may be it's easier to understand and performs better.

Reply:
Done.